src/tag/tag.service.ts

```diff
@@ -5,7 +5,19 @@ import { DbService } from '../db/db.service';
 export class TagService {
   constructor(private readonly prisma: DbService) {}
 
-  async getAllTags() {
-    return await this.prisma.tag.findMany();
+  async searchTags(search: string) {
+    const exactMatch = await this.prisma.tag.findFirst({
+      where: { name: search },
+    });
+    const partialMatch = await this.prisma.tag.findMany({
+      where: { name: { contains: search } },
+      orderBy: { name: 'asc' },
```

Comment:
sort first by popularity, then by name

Reply:
In order to sort by popularity we need to fetch data from `rulesetTag` table and it will be more complex query, do we _really_ need it?

Reply:
Just check [it](https://stackoverflow.com/questions/55830930/ordering-by-multiple-columns-in-prisma)

Reply:
I mean, we don't have `popularity` column in `tag` table, in order to do it we need to count rows with this id in `rulesetTag` table.

Reply:
That's why I hate prisma so much, in normal postgresql I would just create trigger

Reply:
I think you should create `popularity` column on `tag` and increment it manually when tag is added to ruleset or environment